### PR TITLE
Define path with __dir__

### PR DESCRIPTION
**Pull request by DirtyF**
Inspired by https://bogdanvlviv.github.io/posts/ruby/with-__dir__-we-can-restore-order-in-the-universe.html

/cc @bogdanvlviv

### CODE REVIEW

File: lib/jekyll/webmention_io.rb [hunk 1]
```diff
@@ -200,8 +200,8 @@ def self.get_template_contents(template)
         template_file = Jekyll::WebmentionIO.config["templates"][template]
       else
         # Jekyll::WebmentionIO::log 'info', "Using default #{template} template"
-        template_file = File.join(File.dirname(File.expand_path(__FILE__)), "templates/#{template}.html")
-        end
+        template_file = File.join(File.expand_path(__dir__), "templates/#{template}.html")
```
Comment (bogdanvlviv) [inline lib/jekyll/webmention_io.rb]:
```ruby
template_file = File.expand_path("templates/#{template}.html", __dir__)
```

File: lib/jekyll/generators/compile_js.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@ def generate(site)
 
         config = config.merge(site_config)
 
-        source = File.join(File.dirname(File.expand_path(__FILE__)), "../assets/")
+        source = File.join(File.expand_path(__dir__), "assets/")
```
Comment (bogdanvlviv) [inline lib/jekyll/generators/compile_js.rb]:
```ruby
source = File.expand_path("../assets/", __dir__)
```

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+$LOAD_PATH.unshift File.expand_path("../lib", __dir__)
 require "jekyll"
 require "jekyll-webmention_io"
 require "html-proofer"
 
 ENV["JEKYLL_LOG_LEVEL"] = "error"
 
 def dest_dir
-  File.expand_path("../tmp/dest", File.dirname(__FILE__))
+  File.expand_path("tmp/dest", __dir__)
```
Comment (bogdanvlviv) [inline spec/spec_helper.rb]:
```ruby
File.expand_path("../tmp/dest", __dir__)
```

**Comment by DirtyF** [comment]
@bogdanvlviv  addressed your feedback 👍

**Comment by aarongustafson** [comment]
Looks good.

**Pull request merged**